compiler/resolution/functionResolution.cpp

```diff
@@ -7837,9 +7874,6 @@ static void insertReturnTemps() {
               tmp->addFlag(FLAG_MAYBE_TYPE);
               tmp->addFlag(FLAG_MAYBE_PARAM);
             }
-            def->insertAfter(new CallExpr(PRIM_MOVE,
-                                          tmp,
-                                          contextCallOrCall->remove()));
```

Comment:
What's going on with this change?

Reply:
Re shuffling of `def->insertAfter` and the change `contextCallOrCall->insertBefore` to `def->next->insertAfter` :

These changes are necessary because there is now `handleStatementLevelIteratorCall()`, which invokes `resolveForallHeader()` on a `ForallStmt` whose iterable is `tmp`.

For `resolveForallHeader()` to do its job, it needs to see the "definition" (not DefExpr) of `tmp`. So we need to insert that PRIM_MOVE before `handleStatementLevelIteratorCall()`.

The PRIM_MOVE, in turn, requires that `contextCallOrCall` be removed from its original place in the AST. So we can no longer use it to insert `sls` relatively to it. Hence `def->next`.

We can surely achieve the same effect in other ways.
I am open to suggestions.

Reply:
I think it'd be more typical to have an "insertPoint" variable that keeps track of this through the function. I worry that def->next is more subtle/less maintainable. Nonetheless I'm OK with this for now. Thanks for explaining.